src/ui/ViewModelBase.cpp

```diff
@@ -0,0 +1,142 @@
+#include "ViewModelBase.hh"
+
+namespace ra {
+namespace ui {
+
+void ViewModelBase::SetValue(const BoolModelProperty& pProperty, bool bValue)
+{
+    IntModelProperty::ValueMap::const_iterator iter = m_mIntValues.find(pProperty.GetKey());
+
+    if (bValue == pProperty.GetDefaultValue())
+    {
+        if (iter == m_mIntValues.end())
+            return;
+
+        m_mIntValues.erase(iter);
+    }
+    else
+    {
+        if (iter != m_mIntValues.end())
+            return;
+
+        m_mIntValues[pProperty.GetKey()] = static_cast<int>(bValue);
+    }
+
+#ifdef _DEBUG
+    m_mDebugValues[pProperty.GetPropertyName()] = bValue ? L"true" : L"false";
+#endif
+
+    if (!m_vNotifyTargets.empty())
+    {
+        BoolModelProperty::ChangeArgs args{ pProperty, !bValue, bValue };
+
+        // create a copy of the list of pointers in case it's modified by one of the callbacks
+        NotifyTargetSet vNotifyTargets(m_vNotifyTargets);
+        for (NotifyTarget* target : vNotifyTargets)
+            target->OnViewModelBoolValueChanged(args);
```

Comment:
This is how bindings will be notified when a property changes (and _only_ if it changes). Only used by unit tests in this PR.